lib/serializer/BinarySerializer.h

```diff
@@ -364,6 +364,12 @@ class DLL_LINKAGE BinarySerializer : public CSaverBase
 		for(ui32 i = 0; i < length; i++)
 			save(data.data()[i]);
 	}
+	template <std::size_t T>
+	void save(const std::bitset<T> &data)
+	{
+		auto writ = static_cast<uint32_t>(data.to_ulong());
```

Comment:
We can save it as uint64_t, if desired. But I can just add a static_assert, because I cannot imagine why we can need such large bitset. What is better?

Reply:
Or both?
1) add static_assert to make sure that bad code fails to compile instead of producing weird and hard to find bug.
2) serialize based on bitset size as uint16/uint32/uint64 - we have a lot of bonus instances so saving them with as smallest possible size might have notable effect on savegame size.

Reply:
Done with constexpr if